frontend/public/components/catalog-items.jsx

```diff
@@ -25,24 +28,61 @@ export class CatalogTileViewPage extends React.Component {
       allCategories,
       currentCategories: allCategories,
       numItems: _.first(allCategories).numItems,
+      filters: props.filters,
+      activeTabs: ['all'],
+      showAllItemsForCategory: null,
     };
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    const { filters, activeTabs } = this.state;
+
+    if (filters !== prevState.filters) {
+      const allCategories = categorizeItems(this.filterItems());
+
+      if (_.first(activeTabs) === 'all') {
+        this.setState({
+          allCategories,
+          currentCategories: allCategories,
+          numItems: _.first(allCategories).numItems,
+        });
+        return;
+      }
+
+      const mainCategory = _.find(allCategories, { id: _.first(activeTabs) });
+      const curCategories = mainCategory.subcategories || allCategories;
+      const numItems = activeTabs.length < 2 ? mainCategory.numItems : _.find(curCategories, { id: _.last(activeTabs) }).numItems;
+
+      this.setState({
+        allCategories,
+        currentCategories: curCategories,
+        numItems: numItems || 0
+      });
+    }
+  }
+
   isAllTabActive() {
     const { activeTabs } = this.state;
-    return activeTabs[0] === 'all';
+    return _.first(activeTabs) === 'all';
   }
 
   activeTabIsSubCategory(subcategories) {
     const { activeTabs } = this.state;
-    if (_.size(activeTabs) < 2) {
+    if (activeTabs.length < 2) {
       return false;
     }
 
     const activeID = _.last(activeTabs);
     return _.some(subcategories, { id: activeID });
   }
 
+  inActiveTabHierarchy(categoryID) {
+    const { activeTabs } = this.state;
+    const activeParentID = activeTabs.length > 1 ? _.first(activeTabs) : null;
```

Comment:
I think this could just be

```js
const activeParentID = _.first(activeTabs);
```

And you would have the same result (although you might want to rename it something like `activeMainCategoryID`).

Reply:
Not sure that would work.  If current tab is CICD -> Jenkins, then I want CICD and Jenkins to be displayed, not just CICD.

Reply:
> Not sure that would work. If current tab is CICD -> Jenkins, then I want both CICD and Jenkins tabs to be displayed, not just CICD.

I think it would? It should give the same result as before.